src/noshow/api/app_helpers.py

```diff
@@ -99,25 +100,54 @@ def create_treatment_groups(
     if predictions.empty:
         raise ValueError("The predictions DataFrame is empty.")
 
-    # Create prediction score bins using quantile bins, for example 10
-    predictions["score_bin"] = pd.qcut(predictions["prediction"], q=n_bins)
+    # get unique patient ids
+    unique_patient_ids = predictions["pseudo_id"].unique().tolist()
 
+    # Get all unqiue patients from the ApiPatient table and save to df
+    patients = (
+        session.query(ApiPatient).filter(ApiPatient.id.in_(unique_patient_ids)).all()
+    )
+    patients = pd.DataFrame(patients)
+
+    if not patients.empty:
+        # Merge predictions with patients to get treatment group
+        predictions = pd.merge(
+            predictions,
+            patients[["id", "treatment_group"]],
+            right_on="id",
+            left_on="pseudo_id",
+            how="left",
+        )
+    else:
+        predictions["treatment_group"] = None
+
+    # Create prediction score bins using quantile bins
+    predictions["score_bin"] = pd.qcut(predictions["prediction"], q=n_bins)
     predictions = predictions.sort_values(["prediction"])
 
     # only keep top prediction per patient for treatment/control split
     deduplicated = predictions.drop_duplicates(subset="pseudo_id", keep="first")
+    # only keep patients without treatment group
+    deduplicated = deduplicated[deduplicated["treatment_group"].isnull()]
 
     # Create stratified randomization in control and treatment groups
     deduplicated = deduplicated.sort_values(["hoofdagenda", "prediction"])
     deduplicated["treatment_group"] = deduplicated.groupby(
         ["hoofdagenda", "score_bin"], observed=True
     )["prediction"].transform(lambda x: np.arange(len(x)) % 2)
 
-    predictions = pd.merge(
-        predictions,
+    # update predictions with treatment groups from deduplicated based on pseudo_id
+    merged = predictions.merge(
```

Comment:
Bij het consistent houden van de treatment group door een patiënt er standaard in 1 te houden, zou ik controleren hoe veel invloed dit heeft op het formaat/de verhouding van je control en treatment group. Misschien kun je retrospectief met wat data van afgelopen maand kijken.